internal/metrics/system.go

```diff
@@ -0,0 +1,63 @@
+package metrics
+
+import (
+	"github.com/jakub-dzon/k4e-device-worker/internal/configuration"
+	"github.com/jakub-dzon/k4e-operator/models"
+	"reflect"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	systemTargetName                     = "system"
+	DefaultSystemMetricsScrapingInterval = 60
+	NodeExporterMetricsEndpoint          = "http://localhost:9100/metrics"
+)
+
+var defaultSystemMetricsConfiguration = models.SystemMetricsConfiguration{Interval: DefaultSystemMetricsScrapingInterval}
+
+//go:generate mockgen -package=metrics -destination=configuration_provider_mock.go . DeviceConfigurationProvider
+type DeviceConfigurationProvider interface {
+	GetDeviceConfiguration() models.DeviceConfiguration
+}
+
+type SystemMetrics struct {
+	latestConfig atomic.Value
```

Comment:
Just a general question. I see that we use atomic instead of sync. Other locations in code use sync. Are we saying that both are OK for use in the project?

Reply:
Just a thought I had. I think we could have just passed DeviceConfiguration to each feature and had a single update of the entire DeviceCnfiguration. This is instead of having each feature handle its own copy and sync.

Reply:
Yes, they are both OK and both have similar guarantees; I try to use atomic when possible, because it's less hassle than with mutexes.

That's an interesting idea with sharing the configuration. Sometimes the features will need to be informed that a change happened and we might end up with something similar - but maybe not in every case.